lib/internal/crypto/ec.js

```diff
@@ -0,0 +1,297 @@
+      if (err) {
+        return reject(lazyDOMException(
+          'The operation failed for an operation-specific reason',
+          'OperationError'));
+      }
+
+      const algorithm = { name, namedCurve };
+
+      let publicUsages;
+      let privateUsages;
+      switch (name) {
+        case 'ECDSA':
+          publicUsages = getUsagesUnion(usageSet, 'verify');
+          privateUsages = getUsagesUnion(usageSet, 'sign');
+          break;
+        case 'ECDH':
+          publicUsages = [];
+          privateUsages = getUsagesUnion(usageSet, 'deriveKey', 'deriveBits');
+          break;
+      }
+
+      const publicKey =
+        new CryptoKey(
+          pubKey,
+          algorithm,
+          publicUsages,
+          true);
+
+      const privateKey =
+        new CryptoKey(
+          privKey,
+          algorithm,
+          privateUsages,
+          extractable);
+
+      resolve({ publicKey, privateKey });
+    });
+  });
+}
+
+function ecExportKey(key, format) {
+  return jobPromise(new ECKeyExportJob(
+    kCryptoJobAsync,
+    format,
+    key[kKeyObject][kHandle]));
+}
+
+async function ecImportKey(
+  format,
+  keyData,
+  algorithm,
+  extractable,
+  keyUsages) {
+
+  const { name, namedCurve } = algorithm;
+  validateString(namedCurve, 'algorithm.namedCurve');
+  validateOneOf(
+    namedCurve,
+    'algorithm.namedCurve',
+    ObjectKeys(kNamedCurveAliases));
+
+  let keyObject;
+  const usagesSet = new SafeSet(keyUsages);
+  switch (format) {
+    case 'node.keyObject': {
+      if (!isKeyObject(keyData))
+        throw new ERR_INVALID_ARG_TYPE('keyData', 'KeyObject', keyData);
+      if (keyData.type === 'secret')
+        throw lazyDOMException('Invalid key type', 'InvalidAccessException');
+      verifyAcceptableEcKeyUse(name, keyData.type, usagesSet);
+      keyObject = keyData;
+      break;
+    }
+    case 'spki': {
+      verifyAcceptableEcKeyUse(name, 'public', usagesSet);
+      keyObject = createPublicKey({
+        key: keyData,
+        format: 'der',
+        type: 'spki'
+      });
+      break;
+    }
+    case 'pkcs8': {
+      verifyAcceptableEcKeyUse(name, 'private', usagesSet);
+      keyObject = createPrivateKey({
+        key: keyData,
+        format: 'der',
+        type: 'pkcs8'
+      });
+      break;
+    }
+    case 'jwk': {
+      let curve;
+      if (keyData == null || typeof keyData !== 'object')
+        throw lazyDOMException('Invalid JWK keyData', 'DataError');
+
+      if (keyData.kty !== 'EC')
+        throw lazyDOMException('Invalid key type', 'DataError');
+
+      if (keyData.d !== undefined) {
+        verifyAcceptableEcKeyUse(name, 'private', usagesSet);
+      } else {
+        verifyAcceptableEcKeyUse(name, 'public', usagesSet);
+      }
+
+      if (usagesSet.size > 0 &&
+          keyData.use !== undefined &&
+          keyData.use !== 'sig') {
+        throw lazyDOMException('Invalid use type', 'DataError');
+      }
```

Comment:
EC JWKs can be imported for both uses `sig` and `enc`. - When sig they're ECDSA, when enc they're ECDH

Reply:
EC JWKs can be imported for both uses `sig` and `enc`. - When sig they're ECDSA, when enc they're ECDH